tests/model_inheritance/tests.py

```diff
@@ -423,3 +423,35 @@ def test_inheritance_with_same_model_name(self):
     def test_related_name_attribute_exists(self):
         # The Post model doesn't have an attribute called 'attached_%(app_label)s_%(class)s_set'.
         self.assertFalse(hasattr(self.title, 'attached_%(app_label)s_%(class)s_set'))
+
+
+class InheritanceUniqueTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.grand_parent = GrandParent.objects.create(
+            email='grand_parent@example.com',
+            first_name='grand',
+            last_name='parent',
+        )
+
+    def test_unique(self):
+        grand_child = GrandChild(
+            email=self.grand_parent.email,
+            first_name='grand',
+            last_name='child',
+        )
+        self.assertRaises(
```

Comment:
could we check the message too and use the context manager version for better readability?